quickwit/quickwit-query/src/query_ast/wildcard_query.rs

```diff
@@ -0,0 +1,186 @@
+                    .pop()
+                    .with_context(|| "wildcard query generated no term")?;
+                if !tokens.is_empty() {
+                    return Err(InvalidQuery::Other(anyhow::Error::msg(
+                        "wildcard query generated more than one term",
+                    )));
+                }
+                Ok((field, term))
+            }
+            FieldType::JsonObject(json_options) => {
+                let text_field_indexing =
+                    json_options.get_text_indexing_options().ok_or_else(|| {
+                        InvalidQuery::SchemaError(format!(
+                            "field {} is not full-text searchable",
+                            field_entry.name()
+                        ))
+                    })?;
+                let tokenizer_name = text_field_indexing.tokenizer();
+                let analyzer_name = if tokenizer_manager
+                    .get_does_lowercasing(tokenizer_name)
+                    .with_context(|| {
+                        format!("no tokenizer named `{}` is registered", tokenizer_name)
+                    })? {
+                    "lowercase"
+                } else {
+                    "raw"
+                };
+                let mut text_analyzer = tokenizer_manager
+                    .get(analyzer_name)
+                    .with_context(|| "missing built-in tokenizer")?;
+                let mut token_stream = text_analyzer.token_stream(prefix);
+                let mut tokens = Vec::new();
+                let mut term = Term::with_capacity(100);
+                let mut json_term_writer = JsonTermWriter::from_field_and_json_path(
+                    field,
+                    json_path,
+                    json_options.is_expand_dots_enabled(),
+                    &mut term,
+                );
+
+                token_stream.process(&mut |token| {
+                    json_term_writer.set_str(&token.text);
+                    tokens.push(json_term_writer.term().clone());
+                });
+                let term = tokens
+                    .pop()
+                    .with_context(|| "wildcard query generated no term")?;
+                if !tokens.is_empty() {
+                    return Err(InvalidQuery::Other(anyhow::Error::msg(
+                        "wildcard query generated more than one term",
+                    )));
+                }
+                Ok((field, term))
+            }
+            _ => Err(InvalidQuery::SchemaError(
+                "trying to run a Wildcard query on a non-text field".to_string(),
+            )),
+        }
+    }
+}
+
+impl BuildTantivyAst for WildcardQuery {
+    fn build_tantivy_ast_impl(
+        &self,
+        schema: &TantivySchema,
+        tokenizer_manager: &TokenizerManager,
+        _search_fields: &[String],
+        _with_validation: bool,
+    ) -> Result<TantivyQueryAst, InvalidQuery> {
+        // TODO handle escaped wildcard and questionmarks?
```

Comment:
does elasticsearch handle escaped wildcard and `?` ?

Reply:
I checked, and it handle them as being literals, not wildcards

Reply:
If it's not too much work, I would handle that in this PR.